dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/predictor/DefaultPredictionService.java

```diff
@@ -248,7 +248,7 @@ private PredictionSummary predictInternal( Date startDate, Date endDate, List<St
 
                 for ( PredictorGroup predictorGroup : predictorGroupList )
                 {
-                    predictorList.addAll( predictorGroup.getMembers() );
+                    predictorList.addAll( predictorGroup.getSortedMembers() );
```

Comment:
Do you want to sort the `predictorList` again after finished with this for loop ? or they just need to be sorted within their own group.

Reply:
@vietnguyen Thanks, great question. In a scheduled job, the individually-specified predictors, and predictor groups, run in a specified order. So the only thing that the user can't control is the order of the predictors running within a group. So we just want to sort within a group.